### PR TITLE
[6.12.z] Extend test coverage for BZ#2148813

### PR DESCRIPTION
**Pull request by Satellite-QE**
Cherrypick of PR: https://github.com/SatelliteQE/robottelo/pull/10442

Passes locally for 6.12.1, the fix needs to be cherry-picked to 6.13.0.

### CODE REVIEW

File: tests/foreman/api/test_contentmanagement.py [hunk 1]
```diff
@@ -1225,6 +1226,26 @@ def test_positive_sync_container_repo_end_to_end(
             )
             assert result.status == 0
 
+        # Inspect the images with skopeo (BZ#2148813)
+        if not is_open('BZ:2148813'):
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_contentmanagement.py:1230]: @vsedmik As you mentioned this works locally for 6.12.z, so can this is_open block be removed?

File: tests/foreman/api/test_contentmanagement.py [hunk 2]
```diff
@@ -1225,6 +1226,26 @@ def test_positive_sync_container_repo_end_to_end(
...
+            skopeo_cmd = 'skopeo --debug inspect docker://'
+            for path in repo_paths:
+                result = module_capsule_configured.execute(
+                    f'{skopeo_cmd}{target_sat.hostname}/{path}:latest'
+                )
+                assert result.status == 0
+                result = module_capsule_configured.execute(
+                    f'{skopeo_cmd}{module_capsule_configured.hostname}/{path}:latest'
+                )
+                assert result.status == 0
```
Comment (vsedmik) [inline tests/foreman/api/test_contentmanagement.py:1247]:
```suggestion
        result = module_capsule_configured.execute('yum -y install skopeo')
        assert result.status == 0

        target_sat.api.LifecycleEnvironment(
            id=function_lce.id, registry_unauthenticated_pull='true'
        ).update(['registry_unauthenticated_pull'])

        skopeo_cmd = 'skopeo --debug inspect docker://'
        for path in repo_paths:
            result = module_capsule_configured.execute(
                f'{skopeo_cmd}{target_sat.hostname}/{path}:latest'
            )
            assert result.status == 0
            result = module_capsule_configured.execute(
                f'{skopeo_cmd}{module_capsule_configured.hostname}/{path}:latest'
            )
            assert result.status == 0
```

File: tests/foreman/api/test_contentmanagement.py [hunk 3]
```diff
@@ -33,6 +33,7 @@
 from robottelo.content_info import get_repo_files_by_url
 from robottelo.content_info import get_repomd
 from robottelo.content_info import get_repomd_revision
+from robottelo.utils.issue_handlers import is_open
```
Comment (vsedmik) [inline tests/foreman/api/test_contentmanagement.py:36]:
```suggestion
```

**Review by vsedmik — CHANGES_REQUESTED** [review]

**Comment by vsedmik** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_contentmanagement.py::TestCapsuleContentManagement::test_positive_sync_container_repo_end_to_end

**Comment by vsedmik** [comment]
While the test passes locally against RHV (retested once again), the PRT seems to fail consistently against OSP for some reason. Putting on hold for now.

File: tests/foreman/api/test_contentmanagement.py [hunk 4]
```diff
@@ -1225,6 +1225,25 @@ def test_positive_sync_container_repo_end_to_end(
...
+        assert result.status == 0
+
+        target_sat.api.LifecycleEnvironment(
+            id=function_lce.id, registry_unauthenticated_pull='true'
+        ).update(['registry_unauthenticated_pull'])
+
+        skopeo_cmd = 'skopeo --debug inspect docker://'
+        for path in repo_paths:
+            result = module_capsule_configured.execute(
+                f'{skopeo_cmd}{target_sat.hostname}/{path}:latest'
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_contentmanagement.py:1239]: Looking at the PRT failure `level=debug msg="No credentials matching hostname/owxmaghyv-dhgdgwylp-ejgpbagrdl-hhrgdtkwfy-gvsq` it seems it failed due to authentication, so not sure if LCE update for `registry_unauthenticated_pull` worked for capsule or might be capsule sync required after an LCE update?

Comment (vsedmik) [reply tests/foreman/api/test_contentmanagement.py:1239]: Looking into robottelo.log at the PRT run, the same msg was given for the SAT (line 1238), where the status was 0 and stdout provided valid output (it succeeded). While for the CAPS (line 1242) the status was 1 and no stdout (it failed). Locally, both pass well.

**Review by pondrejk — APPROVED** [review]
ack for the code changes

**Comment by JacobCallahan** [comment]
@vsedmik any update on this?

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_contentmanagement.py::TestCapsuleContentManagement::test_positive_sync_container_repo_end_to_end

**Comment by vsedmik** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_contentmanagement.py::TestCapsuleContentManagement::test_positive_sync_container_repo_end_to_end

**Comment by vsedmik** [comment]
We happy @JacobCallahan @Gauravtalreja1 ?

I don't like the sleep in code very much, but it was the only way how to pass on OSP. 
`sleep(10)` works too locally 5/5, for 7 and lower it starts failing, so I used 20 as a "safe" value.

Any ideas?

**Review by JacobCallahan — APPROVED** [review]
Also not a fan of the sleep, but likely the better option

**Review by Griffin-Sullivan — APPROVED** [review]

**Comment by Griffin-Sullivan** [comment]
@vsedmik need to resolve conflicts and we can merge.

**Comment by vsedmik** [comment]
I haven't been able to push to this cherry-picked branch.
Closing in favour of #11217, which resolves the conflict.

**Comment by Gauravtalreja1** [comment]
Thank you @vsedmik

**Pull request closed without merging**